tests/dummy/app/controllers/logging-example/application-log.js

```diff
@@ -0,0 +1,375 @@
+          break;
+        case 'Debug':
+          this.makeDebug();
+          break;
+        case 'Info':
+          this.makeInfo();
+          break;
+        case 'Log':
+          this.makeLog();
+          break;
+        case 'Warn':
+          this.makeWarn();
+          break;
+      }
+    },
+
+    /**
+     *    Close confirm modal dialog.
+     *
+     *    @method hideConfirmModalDialog
+     *    @param {String} confirmCaption Confirm caption (window header caption).
+     *    @param {String} confirmContent Confirm content (window body content).
+     *    @returns {String} Confirm content.
+     */
+    closeDialog() {
+      if (this.confirmModalDialog === undefined) {
+        return;
+      }
+
+      this.set('confirmModalDialogContent', undefined);
+      this.set('confirmModalDialogCategory', undefined);
+      this.confirmModalDialog.modal('hide');
+    }
+
+  },
+
+  makeAssert() {
+    let message = this._getMessageNumber() + 'Assert invocation testing';
+    try {
+      Ember.assert(message, false);
+    } catch (e) {
+      Ember.Logger.error(e);
+      this._router.router.refresh();
+    }
+
+  },
+
+  makeError() {
+    try {
+      eval('error_operator');
+    } catch (e) {
+      e.message = this._getMessageNumber() + e.message;
+      Ember.Logger.error(e);
+      this._router.router.refresh();
+    }
+
+  },
+
+  makeThrow() {
+    try {
+      let message = this._getMessageNumber() + 'Throw invocation testing';
+      throw new Error(message);
+    } catch (e) {
+      Ember.Logger.error(e);
+      this._router.router.refresh();
+    }
+
+  },
+
+  makeDeprecation() {
+    let message = 'DEPRECATION:' + this._getMessageNumber() + 'Deprecation invocation testing';
+    Ember.Logger.warn(message);
+    this._router.router.refresh();
+  },
+
+  makeDebug() {
+    let message = this._getMessageNumber() + 'Debug invocation testing';
+    Ember.Logger.debug(message);
+    this._router.router.refresh();
+  },
+
+  makeInfo() {
+    let message = this._getMessageNumber() + 'Info invocation testing';
+    Ember.Logger.info(message);
+    this._router.router.refresh();
+  },
+
+  makeLog() {
+    let message = this._getMessageNumber() + 'Log invocation testing';
+    Ember.Logger.log(message);
+    this._router.router.refresh();
+  },
+
+  makeWarn() {
+    let message = this._getMessageNumber() + 'Warning invocation testing';
+    Ember.Logger.warn(message);
+    this._router.router.refresh();
+  },
+
+  /**
+   *    Shows confirm modal dialog.
+   *
+   *    @method showConfirmModalDialog
+   *    @param {String} Confirm category
+   *    @returns {String} Confirm content.
+   */
+  showConfirmModalDialog(category) {
+    if (this.confirmModalDialog === undefined) {
+      this.confirmModalDialog = Ember.$('.application-log-modal-dialog');
+      this.confirmModalDialog.modal('setting', 'closable', false);
+    }
+
+    let confirmContent = 'Текущий уровень отладки (' +
+      this.logLevel +
```

Comment:
Construction like '`No projection with '${projectionName}' name defined in '${modelName}' model.`' can be used in order not to concatenate strokes.